test/Tests/src/Network/Http/Request_Spec.enso

```diff
@@ -13,7 +14,7 @@ spec =
     test_headers = [Header.application_json, Header.new "X-Foo-Id" "0123456789"]
     Test.group "Request" <|
         Test.specify "should return error when creating request from invalid URI" <|
-            Request.new HTTP_Method.Post "invalid uri" . should_fail_with Syntax_Error_Data
+            Request.new HTTP_Method.Post "invalid uri" . should_fail_with Syntax_Error.Error
```

Comment:
Nothing new here, but looking at this test I'm not so certain that having one `Syntax_Error` for all kinds of errors is a good idea - a URI parse error is a different thing from a Regex syntax error from ... etc.

Not necessarily to address in this PR, but maybe worth considering - I think we may want to catch a specific kind of error only, not all syntax errors regardless of their origin?